smart-contracts/contracts/tokens/UP/UPGovernor.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.21;
+
+import "@openzeppelin/contracts-upgradeable5/governance/GovernorUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable5/governance/extensions/GovernorSettingsUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable5/governance/extensions/GovernorCountingSimpleUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable5/governance/extensions/GovernorVotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable5/governance/extensions/GovernorTimelockControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable5/proxy/utils/Initializable.sol";
+
+/// @custom:security-contact hello@unlock-protocol.com
+contract UPGovernor is
+  Initializable,
+  GovernorUpgradeable,
+  GovernorSettingsUpgradeable,
+  GovernorCountingSimpleUpgradeable,
+  GovernorVotesUpgradeable,
+  GovernorTimelockControlUpgradeable
+{
+  uint private _quorum;
+
+  // add custom event for quorum changes
+  event QuorumSet(uint oldQuorum, uint newQuorum);
+
+  /// @custom:oz-upgrades-unsafe-allow constructor
+  constructor() {
+    _disableInitializers();
+  }
+
+  function initialize(
+    IVotes _token,
+    TimelockControllerUpgradeable _timelock
+  ) public initializer {
+    __Governor_init("UnlockProtocolGovernor");
+    __GovernorSettings_init(43200 /* 6 day */, 43200 /* 6 days */, 0);
```

Comment:
Is this in seconds? If so this is not 6 days! please use `6* 24 * 60 * 60` for clarty.